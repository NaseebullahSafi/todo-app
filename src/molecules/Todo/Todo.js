import React from "react";
import PropTypes from "prop-types";

import { useTodo } from "src/hooks/useTodo";
import Input from "src/atoms/Input";
import {
  StyledContainer,
  StyledTodo,
  StyledBtn,
  StyledTodoControls,
  StyledDateContainer,
  StyledTodoName,
  StyledTodoDesc,
  StyledTodoFooter,
  StyledTodoAttachments,
  StyledTodoChat,
  StyledSeperator,
  StyledTodoLeft,
  StyledInputContainer,
} from "./Todo.style";

/**
 * Will render the top section of the Todo component
 * @param {Object} controls - an object that carries values needed for the controls
 */
const TodoTopControls = ({ date, editable, onEdit, onDelete, onSubmit }) => {
  return (
    <StyledTodoControls>
      <StyledDateContainer>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="16"
          height="16"
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          className="feather feather-clock"
        >
          <circle cx="12" cy="12" r="10" />
          <polyline points="12 6 12 12 16 14" />
        </svg>
        <span>{date}</span>
      </StyledDateContainer>
      <StyledContainer>
        {editable ? (
          <StyledBtn marginRight check onClick={onSubmit}>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              width="16"
              height="16"
              viewBox="0 0 24 24"
              fill="none"
              stroke="currentColor"
              strokeWidth="2"
              strokeLinecap="round"
              strokeLinejoin="round"
              className="feather feather-check"
            >
              <polyline points="20 6 9 17 4 12" />
            </svg>
          </StyledBtn>
        ) : (
          <StyledBtn marginRight onClick={onEdit}>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              width="16"
              height="16"
              viewBox="0 0 24 24"
              fill="none"
              stroke="currentColor"
              strokeWidth="2"
              strokeLinecap="round"
              strokeLinejoin="round"
              className="feather feather-edit-2"
            >
              <path d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z" />
            </svg>
          </StyledBtn>
        )}
        <StyledBtn remove onClick={onDelete}>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="16"
            height="16"
            viewBox="0 0 24 24"
            fill="none"
            stroke="currentColor"
            strokeWidth="2"
            strokeLinecap="round"
            strokeLinejoin="round"
            className="feather feather-trash-2"
          >
            <polyline points="3 6 5 6 21 6" />
            <path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2" />
            <line x1="10" y1="11" x2="10" y2="17" />
            <line x1="14" y1="11" x2="14" y2="17" />
          </svg>
        </StyledBtn>
      </StyledContainer>
    </StyledTodoControls>
  );
};

/**
 * Will render the footer section of our todo
 */
const TodoFooter = () => (
  <StyledTodoFooter>
    <StyledTodoLeft>
      <StyledTodoAttachments>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="16"
          height="16"
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          className="feather feather-paperclip"
        >
          <path d="M21.44 11.05l-9.19 9.19a6 6 0 0 1-8.49-8.49l9.19-9.19a4 4 0 0 1 5.66 5.66l-9.2 9.19a2 2 0 0 1-2.83-2.83l8.49-8.48" />
        </svg>
        <span>13</span>
      </StyledTodoAttachments>
      <StyledTodoChat>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="16"
          height="16"
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          className="feather feather-message-square"
        >
          <path d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2z" />
        </svg>
        <span>20</span>
      </StyledTodoChat>
    </StyledTodoLeft>

    <StyledBtn circle>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="16"
        height="16"
        viewBox="0 0 24 24"
        fill="none"
        stroke="currentColor"
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
        className="feather feather-plus"
      >
        <line x1="12" y1="5" x2="12" y2="19" />
        <line x1="5" y1="12" x2="19" y2="12" />
      </svg>
    </StyledBtn>
  </StyledTodoFooter>
);

/**
 * Todo component which will render the UI of todo
 * @param {Object} todo - object that defines our todo
 */
const Todo = ({ id, name, description, date }) => {
  // destrcuture our custom hook
  const {
    editable,
    todo,
    formatDate,
    toggleEditable,
    onChange,
    onDelete,
    onSubmit,
  } = useTodo(id, name, description, date);

  return (
    <StyledTodo>
      <TodoTopControls
        date={formatDate(date)}
        editable={editable}
        onEdit={() => toggleEditable(true)}
        onDelete={onDelete}
        onSubmit={onSubmit}
      />
      {editable ? (
        <StyledInputContainer marginTop marginBottom>
          <label htmlFor="name">Todo Name</label>
          <Input
            id="name"
            name="name"
            type="text"
            placeholder="Todo Name"
            value={todo.name}
            onChange={onChange}
          />
        </StyledInputContainer>
      ) : (
        <StyledTodoName>{name}</StyledTodoName>
      )}

      {editable ? (
        <StyledInputContainer marginTop marginBottom>
          <label htmlFor="description">Todo Description</label>
          <Input
            id="description"
            name="description"
            type="text"
            placeholder="Todo Description"
            value={todo.description}
            onChange={onChange}
          />
        </StyledInputContainer>
      ) : (
        <StyledTodoDesc>{description}</StyledTodoDesc>
      )}
      <StyledSeperator />
      <TodoFooter />
    </StyledTodo>
  );
};

Todo.defaultProps = {
  name: "Todo Title",
  description: "Todo Description",
  date: new Date().toString(),
};

Todo.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  description: PropTypes.string,
  date: PropTypes.string,
};

export default Todo;
