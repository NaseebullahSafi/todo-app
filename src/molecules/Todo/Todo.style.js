import styled, { css } from "styled-components";

const StyledFlexRow = css`
  display: flex;
  align-items: center;
`;

export const StyledContainer = styled.div`
  ${StyledFlexRow};
`;

export const StyledTodo = styled.div`
  background-clip: initial;
  background-color: #fff;
  border: 0.0625rem solid rgba(0, 0, 0, 0.05);
  border-bottom-left-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  box-shadow: 0 15px 35px rgba(50, 50, 93, 0.1), 0 5px 15px rgba(0, 0, 0, 0.07) !important;
  display: flex;
  flex: 1 1 auto;
  flex-direction: column;
  margin: 10px;
  max-width: 800px;
  min-width: 0;
  padding: 1.5rem;
  word-wrap: break-word;

  @media screen and (min-width: 300px) {
  }
`;

export const StyledBtn = styled.button`
  ${StyledFlexRow};
  padding: 5px;
  color: #d0d6e2;
  border: 1px dashed #d0d6e2;
  border-radius: 5px;
  transition: all 0.15s ease-in-out;
  margin-right: ${({ marginRight }) => (marginRight ? `1em` : 0)};
  justify-content: center;

  ${({ circle }) =>
    circle
      ? css`
          border-radius: 22px;
          border-top-left-radius: 0;
        `
      : ""}

  ${({ check }) =>
    check
      ? css`
          border-color: transparent;
          background-color: #2dce89;
          svg {
            color: white;
          }
        `
      : ""}

  &:hover {
    border: 1px dashed transparent;
    background-color: ${({ check, remove }) => {
      if (check) return `#26af74`;
      if (remove) return `#f5365c`;
      return `#5e72e4`;
    }};
    color: white;
  }
`;

export const StyledTodoControls = styled.div`
  ${StyledFlexRow};
  justify-content: space-between;
`;

export const StyledDateContainer = styled.div`
  display: flex;
  align-items: center;
  font-size: 12px;
  font-weight: bold;
  color: #d0d6e2;

  svg {
    margin-right: 6px;
  }
`;

export const StyledTodoName = styled.h1`
  font-size: 1em;
  margin: 0.5em 0 0.9em;
`;

export const StyledTodoDesc = styled.p`
  font-size: 0.75em;
  color: #bbc0cc;
  text-align: justify;
  margin: 0;
  line-height: 1.8em;
`;

export const StyledSeperator = styled.hr`
  width: 100%;
  border: 1px dashed #f3f3f3;
  margin: 18px 0;
`;

export const StyledTodoFooter = styled.div`
  ${StyledFlexRow};
  justify-content: space-between;
`;

export const StyledTodoLeft = styled.div`
  ${StyledFlexRow};
`;

export const StyledTodoAttachments = styled.div`
  ${StyledFlexRow};
  justify-content: flex-start;
  color: #d0d6e2;
  font-size: 13px;
  svg {
    margin-right: 4px;
  }
`;

export const StyledTodoChat = styled.div`
  ${StyledFlexRow};
  justify-content: flex-start;
  color: #d0d6e2;
  font-size: 13px;
  margin-left: 1.2em;
  svg {
    margin-right: 4px;
  }
`;

export const StyledInputContainer = styled.div`
  margin-top: ${({ marginTop }) => (marginTop ? `2em` : 0)};
  margin-bottom: ${({ marginBottom }) => (marginBottom ? `2em` : 0)};
`;
