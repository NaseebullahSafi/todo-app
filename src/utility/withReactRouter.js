import React from "react";
import { MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";

export const withReactRouter = (component) =>
  create(<MemoryRouter>{component}</MemoryRouter>);
