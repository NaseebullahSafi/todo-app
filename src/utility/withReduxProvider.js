import React from "react";
import { Provider } from "react-redux";

import store from "../Store";

export const ReduxProvider = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);
