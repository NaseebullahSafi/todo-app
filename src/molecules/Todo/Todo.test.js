import React from "react";
import { shallow, mount } from "enzyme";
import "jest-styled-components";
import { renderHook } from "@testing-library/react-hooks";
import { create } from "react-test-renderer";

import { ReduxProvider } from "../../utility/withReduxProvider";
import reducers from "../../Store/reducer";
import configureStore from "../../Store";
import Todo from "./Todo";

describe("<Todo />", () => {
  const store = configureStore;

  let wrapper;

  beforeEach(() => {
    wrapper = ({ children }) => <ReduxProvider>{children}</ReduxProvider>;
  });

  it("should render Todo", () => {
    const tree = create(
      <ReduxProvider>
        <Todo id="7b2a3db5-af85-42a5-9440-fa4dc6c7e12b" />
      </ReduxProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Redux", () => {
    it("should return initial state of userAction", () => {
      const { userActions } = reducers(undefined, {});
      expect(userActions).toEqual({});
    });

    it("should add todo to todos reducer", () => {
      const { todos } = reducers(
        { todos: {}, userActions: { isRecording: true } },
        {
          type: "TODO_ADD",
          payload: {
            id: "7b2a3db5-af85-42a5-9440-fa4dc6c7e12b",
            name: "1",
            description: "1",
            date: "Sat Apr 25 2020 16:26:23 GMT+0100 (British Summer Time)",
          },
        }
      );
      expect(todos).toEqual({
        "7b2a3db5-af85-42a5-9440-fa4dc6c7e12b": {
          id: "7b2a3db5-af85-42a5-9440-fa4dc6c7e12b",
          name: "1",
          description: "1",
          date: "Sat Apr 25 2020 16:26:23 GMT+0100 (British Summer Time)",
        },
      });
    });

    it("should add the create action to actions stack", () => {
      const { userActions } = reducers(
        {
          todos: {
            "7b2a3db5-af85-42a5-9440-fa4dc6c7e12b": {
              id: "7b2a3db5-af85-42a5-9440-fa4dc6c7e12b",
              name: "1",
              description: "1",
              date: "Sat Apr 25 2020 16:26:23 GMT+0100 (British Summer Time)",
            },
          },
          userActions: { isRecording: true },
        },
        {
          type: "ACTION_ADD_CREATE",
          payload: {
            id: "7b2a3db5-af85-42a5-9440-fa4dc6c7e12b",
            action: "ADD",
          },
        }
      );
      expect(userActions).toEqual({
        isRecording: true,
        actions: [
          {
            id: "7b2a3db5-af85-42a5-9440-fa4dc6c7e12b",
            action: "ADD",
            data: {},
          },
        ],
      });
    });

    it("should edit todo", () => {
      const { todos } = reducers(
        {
          todos: {
            "ff1e494c-e8be-4049-a9dd-606c3d7197de": {
              id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
              name: "Test Todo",
              description: "Test Todo description",
              date: "Sun Apr 26 2020 16:56:52 GMT+0100 (British Summer Time)",
            },
          },
          userActions: {
            isRecording: true,
            actions: [
              {
                id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
                data: {},
                action: "ADD",
              },
            ],
          },
        },
        {
          type: "TODO_EDIT",
          payload: {
            id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
            data: {
              id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
              name: "Test Todo updated",
              description: "Test Todo description",
              date: "Sun Apr 26 2020 16:57:07 GMT+0100 (British Summer Time)",
            },
          },
        }
      );
      expect(todos).toEqual({
        "ff1e494c-e8be-4049-a9dd-606c3d7197de": {
          id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
          name: "Test Todo updated",
          description: "Test Todo description",
          date: "Sun Apr 26 2020 16:57:07 GMT+0100 (British Summer Time)",
        },
      });
    });

    it("should register edit action to our stack", () => {
      const { userActions } = reducers(
        {
          todos: {
            "ff1e494c-e8be-4049-a9dd-606c3d7197de": {
              id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
              name: "Test Todo updated",
              description: "Test Todo description",
              date: "Sun Apr 26 2020 16:57:07 GMT+0100 (British Summer Time)",
            },
          },
          userActions: {
            isRecording: true,
            actions: [
              {
                id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
                data: {},
                action: "ADD",
              },
            ],
          },
        },
        {
          type: "ACTION_ADD_UPDATE",
          payload: {
            id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
            data: {
              id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
              name: "Test Todo updated",
              description: "Test Todo description",
              date: "Sun Apr 26 2020 16:57:07 GMT+0100 (British Summer Time)",
            },
            action: "EDIT",
          },
        }
      );
      expect(userActions).toEqual({
        isRecording: true,
        actions: [
          {
            id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
            data: {},
            action: "ADD",
          },
          {
            id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
            data: {
              id: "ff1e494c-e8be-4049-a9dd-606c3d7197de",
              name: "Test Todo updated",
              description: "Test Todo description",
              date: "Sun Apr 26 2020 16:57:07 GMT+0100 (British Summer Time)",
            },
            action: "EDIT",
          },
        ],
      });
    });
  });
});
