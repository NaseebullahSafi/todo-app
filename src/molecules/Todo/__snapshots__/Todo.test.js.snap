// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> should render Todo 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  background-clip: initial;
  background-color: #fff;
  border: 0.0625rem solid rgba(0,0,0,0.05);
  border-bottom-left-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  box-shadow: 0 15px 35px rgba(50,50,93,0.1),0 5px 15px rgba(0,0,0,0.07) !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 10px;
  max-width: 800px;
  min-width: 0;
  padding: 1.5rem;
  word-wrap: break-word;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px;
  color: #d0d6e2;
  border: 1px dashed #d0d6e2;
  border-radius: 5px;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  margin-right: 1em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4:hover {
  border: 1px dashed transparent;
  background-color: #5e72e4;
  color: white;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px;
  color: #d0d6e2;
  border: 1px dashed #d0d6e2;
  border-radius: 5px;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  margin-right: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5:hover {
  border: 1px dashed transparent;
  background-color: #f5365c;
  color: white;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px;
  color: #d0d6e2;
  border: 1px dashed #d0d6e2;
  border-radius: 5px;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  margin-right: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 22px;
  border-top-left-radius: 0;
}

.c13:hover {
  border: 1px dashed transparent;
  background-color: #5e72e4;
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: bold;
  color: #d0d6e2;
}

.c2 svg {
  margin-right: 6px;
}

.c6 {
  font-size: 1em;
  margin: 0.5em 0 0.9em;
}

.c7 {
  font-size: 0.75em;
  color: #bbc0cc;
  text-align: justify;
  margin: 0;
  line-height: 1.8em;
}

.c8 {
  width: 100%;
  border: 1px dashed #f3f3f3;
  margin: 18px 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  color: #d0d6e2;
  font-size: 13px;
}

.c11 svg {
  margin-right: 4px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  color: #d0d6e2;
  font-size: 13px;
  margin-left: 1.2em;
}

.c12 svg {
  margin-right: 4px;
}

@media screen and (min-width:300px) {

}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <svg
        className="feather feather-clock"
        fill="none"
        height="16"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <polyline
          points="12 6 12 12 16 14"
        />
      </svg>
      <span>
        Apr 26
      </span>
    </div>
    <div
      className="c3"
    >
      <button
        className="c4"
        onClick={[Function]}
      >
        <svg
          className="feather feather-edit-2"
          fill="none"
          height="16"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z"
          />
        </svg>
      </button>
      <button
        className="c5"
        onClick={[Function]}
      >
        <svg
          className="feather feather-trash-2"
          fill="none"
          height="16"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="3 6 5 6 21 6"
          />
          <path
            d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
        </svg>
      </button>
    </div>
  </div>
  <h1
    className="c6"
  >
    Todo Title
  </h1>
  <p
    className="c7"
  >
    Todo Description
  </p>
  <hr
    className="c8"
  />
  <div
    className="c9"
  >
    <div
      className="c10"
    >
      <div
        className="c11"
      >
        <svg
          className="feather feather-paperclip"
          fill="none"
          height="16"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.44 11.05l-9.19 9.19a6 6 0 0 1-8.49-8.49l9.19-9.19a4 4 0 0 1 5.66 5.66l-9.2 9.19a2 2 0 0 1-2.83-2.83l8.49-8.48"
          />
        </svg>
        <span>
          13
        </span>
      </div>
      <div
        className="c12"
      >
        <svg
          className="feather feather-message-square"
          fill="none"
          height="16"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2z"
          />
        </svg>
        <span>
          20
        </span>
      </div>
    </div>
    <button
      className="c13"
    >
      <svg
        className="feather feather-plus"
        fill="none"
        height="16"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="5"
          y2="19"
        />
        <line
          x1="5"
          x2="19"
          y1="12"
          y2="12"
        />
      </svg>
    </button>
  </div>
</div>
`;
